Bundle/ConfigBundle/Favicon/FaviconGenerator.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Victoire\Bundle\ConfigBundle\Favicon;
+
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+use Victoire\Bundle\ConfigBundle\Entity\GlobalConfig;
+
+class FaviconGenerator
+{
+    /**
+     * @var string
+     */
+    public $nodePath;
+    /**
+     * @var string
+     */
+    public $target;
+    /**
+     * @var FaviconConfigDumper
+     */
+    private $faviconConfigDumper;
+    /**
+     * @var string
+     */
+    private $realFaviconPath;
+
+    /**
+     * FaviconGenerator constructor.
+     *
+     * @param FaviconConfigDumper $faviconConfigDumper
+     * @param string              $nodePath
+     * @param string              $realFaviconPath
+     * @param string              $target
+     * @param string              $cwd                 current working directory for process
+     */
+    public function __construct(FaviconConfigDumper $faviconConfigDumper, string $nodePath, string $realFaviconPath, string $target, string $cwd)
+    {
+        $this->nodePath = $nodePath;
+        $this->faviconConfigDumper = $faviconConfigDumper;
+        $this->realFaviconPath = $realFaviconPath;
+        $this->target = $target;
+        $this->cwd = $cwd;
+    }
+
+    /**
+     * Generate favicons from realfavicon generator and a GlobalConfig object.
+     *
+     * @param GlobalConfig $globalConfig
+     * @param string       $path
+     *
+     * @return array The generated files
+     */
+    public function generate(GlobalConfig $globalConfig, $path = 'faviconConfig.json')
+    {
+        return $this->generateFromConfigFile($this->faviconConfigDumper->dump($globalConfig, $path));
+    }
+
+    /**
+     * Generate favicons from a config file.
+     *
+     * @param string $configPath
+     *
+     * @throws GenerateFaviconException
+     *
+     * @return array The generated files
+     */
+    public function generateFromConfigFile(string $configPath)
+    {
+        try {
+            $fileSystem = new Filesystem();
+            if (!$fileSystem->exists($configPath)) {
+                throw new FileNotFoundException(sprintf('The file %s does not exist.', $configPath));
+            }
+
+            $faviconRequestPath = $this->cwd.'/faviconRequest.json';
+            $fileSystem = new Filesystem();
+            if (!$fileSystem->exists($faviconRequestPath)) {
+                $fileSystem->copy($configPath, $faviconRequestPath);
+            }
+            $this->process = new Process(sprintf(
```

Comment:
Why store process in a class variable ? `$process` would avoid confusion with a potential stateful service